### PR TITLE
SensitivityTensor

### PR DESCRIPTION
**Pull request by iamtrask**
This pull request is the initial merging of the SensitivityTensor project.

### CODE REVIEW

File: syft/frameworks/torch/tensors/decorators/sensitivity.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+            if_left = other <= self.max_vals
+            if_right = other >= self.min_vals
+            if_and = if_left * if_right
+
+            new_max_val = if_and
+            new_min_val = new_max_val * 0
+
+        return SensitivityTensor(new_vals, new_max_val, new_min_val)
+
+    def __lt__(self, other):
```
Comment (robert-wagner) [inline syft/frameworks/torch/tensors/decorators/sensitivity.py]: Can we define less than in terms of gt or vice versa to better reuse code?

File: syft/frameworks/torch/tensors/decorators/sensitivity.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+
+            if_left = other <= self.max_vals
+            if_right = other >= self.min_vals
+            if_and = if_left * if_right
+
+            new_max_val = if_and
+            new_min_val = new_max_val * 0
+
+        return SensitivityTensor(result, new_max_val, new_min_val)
+
```
Comment (robert-wagner) [inline syft/frameworks/torch/tensors/decorators/sensitivity.py]: Should we have an equals method?

File: syft/frameworks/torch/tensors/decorators/sensitivity.py [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+    def clamp_min(self, other):
```
Comment (robert-wagner) [inline syft/frameworks/torch/tensors/decorators/sensitivity.py]: Should this and clamp_max be private?

File: syft/frameworks/torch/tensors/decorators/sensitivity.py (hunk 1 above)
Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: erm - maybe - i'll have to get back to you on that.

Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: we can definitely get at least come code reuse there - maybe break out the internals into a helper method or two

Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: the min/max manipulation is very similar between the two

Comment (robert-wagner) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: I'm pretty sure gt is just lt with self and other switched

File: syft/frameworks/torch/tensors/decorators/sensitivity.py (hunk 3 above)
Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: nope

Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: https://pytorch.org/cppdocs/api/function_namespaceat_1ab838000b4810bff416dce47af5604d3b.html

File: syft/frameworks/torch/tensors/decorators/sensitivity.py (hunk 2 above)
Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: +1 

TODO

File: syft/frameworks/torch/tensors/decorators/sensitivity.py (hunk 1 above)
Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: oh... fair point

Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: done

File: syft/frameworks/torch/tensors/decorators/sensitivity.py (hunk 3 above)
Comment (robert-wagner) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: Huh. Neat

File: syft/frameworks/torch/tensors/decorators/sensitivity.py [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+            raise Exception("probably best not to do this - it's gonna be inf a lot")
+
+        new_vals = self.values / other
+        new_max_vals = self.max_vals / other
+        new_min_vals = self.min_vals / other
+
+        return SensitivityTensor(new_vals, new_max_vals, new_min_vals)
+
+    def __gt__(self, other):
+        """BUG: the zero values mess this up"""
```
Comment (robert-wagner) [inline syft/frameworks/torch/tensors/decorators/sensitivity.py]: Did lt also how the zero values issue?

File: syft/frameworks/torch/tensors/decorators/sensitivity.py (hunk 2 above)
Comment (iamtrask) [reply syft/frameworks/torch/tensors/decorators/sensitivity.py]: done

**Pull request closed without merging**